app/models/user.rb

```diff
@@ -45,17 +29,51 @@ class User < ApplicationRecord
 
   GRADES = %w[9 10 11 12 ផ្សេងៗ].freeze
 
-  HOUSE_TYPES = %w[ផ្ទះឈើ ផ្ទះឈើលើថ្មក្រោម ផ្ទះថ្ម ផ្ទះស័ង្កសី ផ្ទះស្លឹក].freeze
-
+  belongs_to :high_school, foreign_key: :high_school_code, optional: true
   has_many :games
   has_many :personal_understandings, through: :games
 
   validates :grade, inclusion: { in: GRADES }, allow_nil: true
-  validates :house_type, inclusion: { in: HOUSE_TYPES }, allow_nil: true
+
+  def address
+    return nil if high_school_code.blank?
+    district = high_school.location
+    "#{commune.try(:name_km)} #{district.name_km} #{district.province.name_km}"
+  end
+
+  def commune
+    Pumi::Commune.find_by_id(commune_code)
+  end
+
+  def district
+    Pumi::District.find_by_id(district_code)
+  end
+
+  def province
+    Pumi::Province.find_by_id(province_code)
+  end
+
+  def high_school_label
+    return nil if high_school_code.blank?
+
+    "#{province.name_km} > #{district.name_km} > #{high_school.name_km}"
+  end
+
+  def self.grades
+    [
+      { value: 9, label: 9},
+      { value: 10, label: 10},
+      { value: 11, label: 11},
+      { value: 12, label: 12},
+      { value: 'other', label: 'ផ្សេងៗ'}
```

Comment:
@hengsokly for option 'other' seems inconsistent with field grade validation 'ផ្សេងៗ'